gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -100,125 +100,538 @@ var fluid = fluid || require("infusion");
+        var input = {
+            gpiiToken: gpiiToken,
+            oauth2ClientId: oauth2ClientId,
+            selectedPreferences: selectedPreferences
+        };
+        return fluid.promise.fireTransformEvent(that.events.onAddAuthorization, input);
+    };
+
+    // Shared by getSelectedPreferences() and setSelectedPreferences()
+    // @dataStore (Object): a data store instance
+    // @input (Object): The structure is,
+    // {
+    //     gpiiToken: (String),
+    //     oauth2ClientId: (String),
+    //     selectedPreferences: (Object)
+    // }
+    // @return (Promise): a promise object to be passed to the next processing function
+    gpii.oauth2.authorizationService.findGpiiToken = function (dataStore, input) {
+        var promiseTogo = fluid.promise();
+        var gpiiTokenPromise = dataStore.findGpiiToken(input.gpiiToken);
+
+        gpiiTokenPromise.then(function (gpiiToken) {
+            var combined = {
+                gpiiToken: gpiiToken,
+                inputArgs: input
+            };
+            promiseTogo.resolve(combined);
+        }, function (err) {
+            promiseTogo.reject(err);
+        });
+        return promiseTogo;
+    };
+
+    // @dataStore (Object): a data store instance
+    // @record (Object): The structure is,
+    // {
+    //     gpiiToken: {Object},   // A GPII token record
+    //     inputArgs: {
+    //         gpiiToken: (String),
+    //         oauth2ClientId: (String),
+    //         selectedPreferences: (Object)
+    //     }
+    // }
+    // @return (Promise): a promise object to be passed to the next processing function
+    gpii.oauth2.authorizationService.findClient = function (dataStore, record) {
+        var promiseTogo = fluid.promise();
+        var clientPromise = dataStore.findClientByOauth2ClientId(record.inputArgs.oauth2ClientId);
+        clientPromise.then(function (client) {
+            var combined = fluid.extend({}, record, {client: client});
+            promiseTogo.resolve(combined);
+        }, function (err) {
+            promiseTogo.reject(err);
+        });
+        return promiseTogo;
+    };
+
+    // @dataStore (Object): a data store instance
+    // @record (Object): The structure is,
+    // {
+    //     gpiiToken: {Object},   // A GPII token record
+    //     client: {Object},    // A client record
+    //     inputArgs: {
+    //         gpiiToken: (String),
+    //         oauth2ClientId: (String),
+    //         selectedPreferences: (Object)
+    //     }
+    // }
+    // @return (Promise): a promise object. At success, returns the promise object with value in this structure:
+    // {
+    //     id: (String)
+    // }
+    // The returned structure may contain more fields if the promise is returned by dataStore.addAuthDecision()
+    gpii.oauth2.authorizationService.doAdd = function (dataStore, codeGenerator, record) {
```

Comment:
This function has become highly redundant with gpii.oauth2.authorizationService.checkAuthDecision - refactor to a common utility